app/utils/callback/lava.py

```diff
@@ -470,6 +530,8 @@ def _add_test_results(group, results):
         }
         test_case.update({k: test[v] for k, v in TEST_CASE_MAP.iteritems()})
         test_case.update({k: group[k] for k in TEST_CASE_GROUP_KEYS})
+        log_line_data.append((test_case[models.NAME_KEY],
```

Comment:
One issue with this is that several test cases can have the same name within a same job, in different sub-groups.  The `log_line_data` structure is global for all the tests, so the `test_case_path` string should be used instead of the `name`.

Also I wonder if it would not be better to leave this function as-is and have a separate function just to construct the log_line_data.  I know it would mean traversing the structure one more time, but this should be very fast and it would help keeping the purpose of this `_add_test_results` function.

Reply:
I deliberately broke the separation of concerns here to not traverse the same structure twice, but on second thought it may be better to add a separate function to add the log lines and do not prematurely optimize the code.

Nice catch with the `name` vs. `test_case_path`. 

I'll fix the issue.